tools/testbench/topology.c

```diff
@@ -407,29 +432,39 @@ static int tb_register_filewrite(struct testbench_prm *tp,
 			tp->output_file_index);
 		return -EINVAL;
 	}
-	filewrite.fn = strdup(tp->output_file[tp->output_file_index]);
+	filewrite->fn = strdup(tp->output_file[tp->output_file_index]);
 	if (tp->output_file_index == 0)
 		tp->fw_id = ctx->comp_id;
 	tp->output_file_index++;
 
 	/* Set format from testbench command line*/
-	filewrite.rate = tp->fs_out;
-	filewrite.channels = tp->channels_out;
-	filewrite.frame_fmt = tp->frame_fmt;
-	filewrite.direction = dir;
-
-	/* Set type depending on direction */
-	filewrite.comp.type = (dir == SOF_IPC_STREAM_PLAYBACK) ?
-		SOF_COMP_DAI : SOF_COMP_HOST;
+	filewrite->rate = tp->fs_out;
+	filewrite->channels = tp->channels_out;
+	filewrite->frame_fmt = tp->frame_fmt;
+	filewrite->direction = dir;
+
+	file_uuid = (struct sof_uuid *)((uint8_t *)filewrite + sizeof(struct sof_ipc_comp_file));
+	file_uuid->a = 0xbfc7488c;
+	file_uuid->b = 0x75aa;
+	file_uuid->c = 0x4ce8;
+	file_uuid->d[0] = 0x9d;
+	file_uuid->d[1] = 0xbe;
+	file_uuid->d[2] = 0xd8;
+	file_uuid->d[3] = 0xda;
+	file_uuid->d[4] = 0x08;
+	file_uuid->d[5] = 0xa6;
+	file_uuid->d[6] = 0x98;
+	file_uuid->d[7] = 0xc2;
```

Comment:
This was commented by Liam in #9025, is there a nicer way to set this with a macro?

Reply:
define a `static const` UUID and `memcpy()` it?

Reply:
Yep, that works, thanks!